frontend/src/components/viewers/MarkdownViewer.test.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import { mount, ReactWrapper } from 'enzyme';
+import MarkdownViewer, { MarkdownViewerConfig } from './MarkdownViewer';
+import { PlotType } from './Viewer';
+
+describe('MarkdownViewer', () => {
+  let tree: ReactWrapper<any> | null;
+
+  afterEach(() => {
+    if (tree) {
+      tree.unmount();
+      tree = null;
+    }
+  });
+
+  it('does not break on empty data', () => {
+    tree = mount(<MarkdownViewer configs={[]} />);
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders some basic markdown', () => {
+    const markdown = '# Title\n[some link here](http://example.com)';
+    const config: MarkdownViewerConfig = {
+      markdownContent: markdown,
+      type: PlotType.MARKDOWN,
+    };
+
+    tree = mount(<MarkdownViewer configs={[config]} />);
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('sanitizes the markdown to remove XSS', () => {
+    const markdown = `
+  lower[click me](javascript&#x3a;...)lower
+  upper[click me](javascript&#X3a;...)upper
+  `;
```

Comment:
Is this the correct way of writing your strings? Or perhaps something that can follow your tab convention like:
```js
const markdown = [
  'lower[click me](javascript&#x3a;...)lower'
  'upper[click me](javascript&#X3a;...)upper'
].join('\n')
```

Reply:
Since white space doesn't really matter to the test, I'll just indent the strings to match the general style.